Extension/test/scenarios/SingleRootProject/tests/relatedFilesProvider.test.ts

```diff
@@ -0,0 +1,305 @@
+        };
+        mockCopilotApi = sinon.createStubInstance(MockCopilotApi);
+
+        activeClientStub = sinon.createStubInstance(DefaultClient);
+        getActiveClientStub = sinon.stub(extension, 'getActiveClient').returns(activeClientStub);
+        activeClientStub.getIncludes.resolves({ includedFiles: [] });
+    });
+
+    afterEach(() => {
+        sinon.restore()
+    });
+
+    it('should not register provider if CppToolsRelatedFilesApi is not enabled', async () => {
+        getIsRelatedFilesApiEnabledStub.withArgs('CppToolsRelatedFilesApi').resolves(false);
+
+        await moduleUnderTest.registerRelatedFilesProvider();
+
+        ok(getIsRelatedFilesApiEnabledStub.calledOnce, 'getIsRelatedFilesApiEnabled should be called once');
+    });
+
+    it('should register provider if CppToolsRelatedFilesApi is enabled', async () => {
+        getIsRelatedFilesApiEnabledStub.withArgs('CppToolsRelatedFilesApi').resolves(true);
+        const vscodeGetExtensionsStub = sinon.stub(vscode.extensions, 'getExtension').returns({
+            id: 'test-extension-id',
+            extensionUri: vscode.Uri.parse('file:///test-extension-path'),
+            extensionPath: 'test-extension-path',
+            isActive: true,
+            packageJSON: { name: 'test-extension-name' },
+            activate: async () => { },
+            exports: mockCopilotApi,
+            extensionKind: vscode.ExtensionKind.UI
+        });
+
+        await moduleUnderTest.registerRelatedFilesProvider();
+
+        ok(getIsRelatedFilesApiEnabledStub.calledOnce, 'getIsRelatedFilesApiEnabled should be called once');
+        ok(vscodeGetExtensionsStub.calledOnce, 'vscode.extensions.getExtension should be called once');
+        ok(mockCopilotApi.registerRelatedFilesProvider.calledThrice, 'registerRelatedFilesProvider should be called three times');
+        ok(mockCopilotApi.registerRelatedFilesProvider.calledWithMatch(sinon.match({ extensionId: 'test-extension-id', languageId: sinon.match.in(['c', 'cpp', 'cuda-cpp']) })), 'registerRelatedFilesProvider should be called with the correct providerId and languageId');
+    });
+
+    it('should handle provider requests properly when ChatContext isn\'t available.', async () => {
+        getIsRelatedFilesApiEnabledStub.withArgs('CppToolsRelatedFilesApi').resolves(true);
+        const vscodeGetExtensionsStub = sinon.stub(vscode.extensions, 'getExtension').returns({
+            id: 'test-extension-id',
+            extensionUri: vscode.Uri.parse('file:///test-extension-path'),
+            extensionPath: 'test-extension-path',
+            isActive: true,
+            packageJSON: { name: 'test-extension-name' },
+            activate: async () => { },
+            exports: mockCopilotApi,
+            extensionKind: vscode.ExtensionKind.UI
+        });
+        activeClientStub.getIncludes.resolves({ includedFiles: ['c:\\system\\include\\vector', 'c:\\system\\include\\string', 'C:\\src\\my_project\\foo.h'] });
```

Comment:
Can you add a test case with 2 headers for bug https://github.com/microsoft/vscode-cpptools/issues/12807? Or should that be added after that bug is fixed?

Reply:
Tried it locally but couldn't repro bug #12807, note that I mocked `Client.getIncludes` in my tests.

![image](https://github.com/user-attachments/assets/9750e682-e9b9-4dfd-a597-c4859c95f8d1)

![image](https://github.com/user-attachments/assets/47d71caa-bc37-4d15-8db4-34e8ff7431c9)